ui.apps/src/main/content/jcr_root/apps/core/cif/components/content/teaser/v2/teaser/clientlib/editor/js/teaser.js

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ *
+ *    Copyright 2020 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+class TeaserConfig {
+    constructor(jQuery) {
+        this.$ = jQuery;
+        this.handleDialogLoaded = this.handleDialogLoaded.bind(this);
+        this.attachEventHandlers = this.attachEventHandlers.bind(this);
+        this.setFieldsDisabled = this.setFieldsDisabled.bind(this);
+        this.actionsToggleHandler = this.actionsToggleHandler.bind(this);
+        this.handlePickersChange = this.handlePickersChange.bind(this);
+        this.handlePageChange = this.handlePageChange.bind(this);
+        this.handleProductChange = this.handleProductChange.bind(this);
+        this.handleCategoryChange = this.handleCategoryChange.bind(this);
+
+        // Listening for dialog windows to open
+        // The config inputs are available only when the right dialog opens
+        this.$(document).on('dialog-loaded', this.handleDialogLoaded);
```

Comment:
Is this some new code? If so, we shouldn't rely on jQuery anymore unless is absolutely necessary (CoralUI interop, like `adaptTo()`). 
`document.addEventListener` works just as well here.

Reply:
It is taken from V1 of the component. Since V2 directly extends WCM core component and not our V1 I didn't want to reuse the V1 clientlib to not introduce any dependencies we don't need.